### PR TITLE
Add icons for new Jetpack products

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR adds the icons of the new Jetpack products to the Calypso codebase.

### Implementation notes

I appended `v2` to all new slugs and file names for now to identify the new ones easily.

### Testing instructions

- Download the PR
- Go to `devdocs/design/product-icon`
- Check that you see the new icons (see screenshot below)

#### Devdocs
Screen Shot 2020-08-05 at 11 41 20 AM

#### Mockups
Screen Shot 2020-08-05 at 11 41 37 AM

### CODE REVIEW

File: packages/components/src/product-icon/README.md [hunk 1]
```diff
@@ -13,15 +12,14 @@ If you're using a custom Webpack environment, make sure that you have the [asset
...
 
 export default function MyComponent() {
-    return (
-        <div className="my-component">
-            <ProductIcon slug="plan_free" />
-            <ProductIcon slug="jetpack_business" />
-            <ProductIcon slug="jetpack_backup_daily_monthly" />
-        </div>
-    );
+	return (
```
Comment (monsieur-z) [inline packages/components/src/product-icon/README.md:15]: Just some formatting.

File: packages/components/src/product-icon/README.md [hunk 2]
```diff
@@ -68,11 +90,12 @@ Additional class name to be added to the icon element.
...
 In order to add a new icon to the component, do the following:
-* Add a new icon image to the `images` folder.
-* Import the image in the `config.js` file. This way an image URL will be generated by a bundler for further reuse.
-* Add the image path to the `paths` object in the `config.js` file.
-* Add a new entry in the `iconToProductSlugMap` object in the `config.js` file:
-  * key should match the image slug that just have been added,
-  * value should be an array of product slugs the image should be displayed for.
-* Document the new icon as an available option for the `slug` prop in this README.
-* Optionally, update the `docs/example.jsx` file to include the new icon.
+
```
Comment (monsieur-z) [inline packages/components/src/product-icon/README.md:93]: Just some formatting.

File: packages/components/src/product-icon/README.md [hunk 3]
```diff
@@ -1,5 +1,4 @@
-ProductIcon Component
-=============
+# ProductIcon Component
```
Comment (monsieur-z) [inline packages/components/src/product-icon/README.md:1]: Just some formatting.

**Comment by nsakaimbo** [comment]
👋 Friendly heads up: the type of error in the wp-desktop-source job is an intermittent out-of-memory issue that's become more frequent over the last couple of days. A re-try should help, although I just pushed a PR to master to mitigate this (i.e. bump the RAM available to the desktop job) in #44675. Hopefully that should hopefully stabilize things going forward. Apologies for the noise!

File: packages/components/src/product-icon/config.js [hunk 4]
```diff
@@ -18,14 +24,20 @@ import wpcomPremium from './images/wpcom-premium.svg';
...
 	'jetpack-scan': jetpackScan,
+	'jetpack-scan-v2': jetpackScanV2,
 	'jetpack-backup-daily': jetpackBackup,
 	'jetpack-backup-realtime': jetpackBackup,
+	'jetpack-backup-v2': jetpackBackupV2,
 	'jetpack-free': jetpackFree,
 	'jetpack-personal': jetpackPersonal,
 	'jetpack-premium': jetpackPremium,
 	'jetpack-professional': jetpackProfessional,
+	'jetpack-all-in-v2': jetpackAllInV2,
```
Comment (rcanepa) [inline packages/components/src/product-icon/config.js:35]: We might have to come back to this one later and change it to Complete if we stick with the latter as the final name of the bundle.

Comment (monsieur-z) [reply packages/components/src/product-icon/config.js:35]: Absolutely. I believe the deadline to settle on a name is tomorrow. I'll keep an eye on it.

**Pull request merged**